frontend/app/players/page.js

```diff
@@ -0,0 +1,198 @@
+'use client';
+
+import { useState } from 'react';
+import useRequest from '@/utils/useRequest';
+import Modal from 'react-bootstrap/Modal';
+import Button from 'react-bootstrap/Button';
+import OverlayTrigger from 'react-bootstrap/OverlayTrigger';
+import Tooltip from 'react-bootstrap/Tooltip';
+
+export default function Players() {
+  const { response: players, error: playersError, loading: playersLoading, refetch } = useRequest('GET', '/api/players');
+  const { response: matches, error: matchesError, loading: matchesLoading } = useRequest('GET', '/api/matches');
+  const [playerName, setPlayerName] = useState('');
+  const [playerRequested, setPlayerRequested] = useState(null);
+  const playerStats = (!playersLoading && !matchesLoading) && calculatePlayerStats(players, matches);
+
+  async function AddPlayer(playerName) {
+    if (playerName === '') {
+      return;
+    }
+
+    const url = '/api/players';
+    let duplicate = false;
+
+    players.forEach((player) => {
+      const { pname } = player;
+      if (pname.toLowerCase() === playerName.toLowerCase()) {
+        duplicate = true;
+      }
+    });
+
+    if (duplicate) {
+      setPlayerRequested('error');
+      return;
+    }
+
+    try {
+      const response = await fetch(url, {
+        method: 'POST',
+        body: JSON.stringify({ 'pname': playerName }),
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      });
+
+      if (!response.ok) {
+        throw new Error(`Response status: ${response.status}`);
+      }
+
+      refetch();
+      setPlayerRequested('success');
+    } catch (error) {
+      console.error(error.message);
+    };
+  }
+
+  if (!playersError && !playersLoading && !matchesError && !matchesLoading) {
```

Comment:
lets make 2 if statements above that returns some loading component if `playersLoading` or `matchesLoading`
and returns some error message to the page on error (along with the console.log)